chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -113,6 +113,12 @@
 
 nova = search(:node, "run_list_map:nova-controller").first
 
+unless nova.nil?
+  keystone_settings = KeystoneHelper.keystone_settings(nova, "nova")
+  nova_insecure = node[:nova][:ssl][:enabled] && node[:nova][:ssl][:insecure]
+  nova_insecure = keystone_settings["insecure"] || nova_insecure
```

Comment:
I think ```nova_insecure = keystone_settings["insecure"] || node[:nova][:ssl][:insecure]``` would be sufficient here. (Well that's ignoring my other comment about keystone_settings being wrong)